src/Npgsql/NpgsqlConnector.cs

```diff
@@ -2070,7 +2070,19 @@ internal async Task<bool> Wait(bool async, int timeout, CancellationToken cancel
 
                 while (true)
                 {
-                    var msg = await ReadMessageWithoutCancellation(async, cancellationToken);
+                    IBackendMessage? msg;
+
+                    try
+                    {
+                        msg = await ReadMessageWithNotifications(async, cancellationToken);
+                    }
+                    catch (Exception e) when (!IsBroken)
+                    {
+                        // We're somewhere in the middle of a reading keepalive messages
```

Comment:
Which kind of error/scenario exactly is this supposed to address?

For unexpected message, we have a check below (`msg.Code != expectedMessageCode`) where Break is indeed missing... All other kinds of errors (e.g. network) should be handled lower down the stack within ReadMessageWithNotifications

Reply:
If there was a timeout or cancellation through `CancellationToken`, `ReadBuffer` wouldn't break the connection (because ReadMessageWithNotifications sets the `readingNotifications` flag). We can add another, but I'm not sure it's a great idea.

Reply:
OK, I guess I'm fine with it. We could narrow the exceptions caught - mostly for documentation reasons - but up to you.

Also, checking IsBroken shouldn't be necessary, since it's always OK to call Break again (which checks internally).